webapp/channels/src/components/spinner_button.tsx

```diff
@@ -1,37 +1,35 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {PureComponent} from 'react';
-import type {ButtonHTMLAttributes, ReactNode} from 'react';
+import React from 'react';
+import type {ReactNode} from 'react';
 
 import LoadingWrapper from 'components/widgets/loading/loading_wrapper';
 
-type Props = {
+type Props = React.ButtonHTMLAttributes<HTMLButtonElement> & {
     children?: ReactNode;
     spinning: boolean;
     spinningText: ReactNode;
 }
 
-export default class SpinnerButton extends PureComponent<Props & ButtonHTMLAttributes<HTMLButtonElement>> {
-    public static defaultProps: Partial<Props> = {
-        spinning: false,
-    };
+const SpinnerButton = ({spinning, spinningText, children, ...props}: Props) => {
```

Comment:
Then the number of parameters grow, this can be hard to manage, specially when we add defaults. Can you make this multiline?
```suggestion
const SpinnerButton = ({
    spinning,
    spinningText,
    children,
    ...props,
}: Props) => {
```